js/ui/grid_core/ui.grid_core.virtual_scrolling.js

```diff
@@ -671,16 +683,35 @@ const VirtualScrollingRowsViewExtender = (function() {
         },
 
         setLoading: function(isLoading, messageText) {
-            const that = this;
-            const callBase = that.callBase;
-            const dataController = that._dataController;
-            const hasBottomLoadPanel = dataController.pageIndex() > 0 && dataController.isLoaded() && !!that._findBottomLoadPanel();
+            const dataController = this._dataController;
+            const hasBottomLoadPanel = dataController.pageIndex() > 0 && dataController.isLoaded() && !!this._findBottomLoadPanel();
+            const operationTypes = dataController.dataSource()?.operationTypes?.();
+            const hasEnabledOperationType = function(types) {
+                let result = false;
+                if(types) {
+                    for(const t in types) {
+                        if(types[t]) {
+                            result = true;
+                            break;
+                        }
+                    }
+                }
+
+                return result;
+            };
+
+            if(this.option(NEW_SCROLLING_MODE) && isLoading && operationTypes) {
+                const loadingOperationTypes = dataController.dataSource()?.loadingOperationTypes?.();
+                if(loadingOperationTypes?.paging || loadingOperationTypes?.fullReload || !hasEnabledOperationType(loadingOperationTypes)) {
```

Comment:
fullReload should be removed in the future because it exists on refresh

Reply:
Yes, I'm planning to reconsider this condition after improving loading logic.